core/src/main/java/ai/timefold/solver/core/api/solver/SolverJobBuilder.java

```diff
@@ -81,6 +81,14 @@ default SolverJobBuilder<Solution_, ProblemId_> withProblem(Solution_ problem) {
     SolverJobBuilder<Solution_, ProblemId_>
             withFirstInitializedSolutionConsumer(Consumer<? super Solution_> firstInitializedSolutionConsumer);
 
+    /**
+     * Sets the runnable action for when the solver starts its solving process.
+     *
+     * @param startSolverJobHandler never null, called only once when the solver is starting the solving process
+     * @return this, never null
+     */
+    SolverJobBuilder<Solution_, ProblemId_> withStartSolverJobHandler(Runnable startSolverJobHandler);
```

Comment:
I'm wondering... how do I find out which job is starting? Maybe we want to replace `Runnable` with something that can give some context? Not sure.